libraries/src/Console/JobListCommand.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * Joomla! Content Management System
+ *
+ * @copyright  Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\CMS\Console;
+
+defined('JPATH_PLATFORM') or die;
+
+use Joomla\CMS\Factory;
+use Joomla\Console\Command\AbstractCommand;
+use Joomla\Database\DatabaseInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Console command for list jobs queue
+ *
+ * @since  4.0.0
+ */
+class JobListCommand extends AbstractCommand
+{
+	/**
+	 * The default command name
+	 *
+	 * @var    string
+	 * @since  4.0.0
+	 */
+	protected static $defaultName = 'job:list';
+
+	/**
+	 * The elapsed time
+	 *
+	 * @var    string
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	private $time;
+
+	/**
+	 * Database connector
+	 *
+	 * @var    DatabaseInterface
+	 * @since  __DEPLOY_VERSION__
+	 */
+	private $db;
+
+	/**
+	 * Instantiate the command.
+	 *
+	 * @param   DatabaseInterface  $db  Database connector
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function __construct(DatabaseInterface $db)
+	{
+		$this->db = $db;
+		parent::__construct();
+	}
+
+	/**
+	 * Internal function to execute the command.
+	 *
+	 * @param   InputInterface   $input   The input to inject into the command.
+	 * @param   OutputInterface  $output  The output to inject into the command.
+	 *
+	 * @return  integer  The command exit code
+	 *
+	 * @since   4.0.0
+	 */
+	protected function doExecute(InputInterface $input, OutputInterface $output): int
+	{
+		$symfonyStyle = new SymfonyStyle($input, $output);
+
+		$symfonyStyle->title('Job list');
+
+		// Initialize the time value.
+		$this->time = microtime(true);
+
+		$now  = time();
+		$list = $this->listJobs();
+
+		foreach ($list as $job)
+		{
+			$nextrun    = ', next run will be at the next schedule';
+			$taskParams = json_decode($job->params, true);
+			$lastrun    = $taskParams['lastrun'];
+			$runned     = $taskParams['taskid'];
+			$timeout    = $taskParams['cachetimeout'];
+			$unit       = $taskParams['unit'];
+			$timeout    = ($unit * $timeout);
+
+			if ((abs($now - $lastrun) < $timeout))
+			{
+				$nextrun = ', next run will be after ' . abs($now - $lastrun - $timeout);
+			}
+
+			$symfonyStyle->note('Job: ' . $job->element . ' runned ' . $runned . ' times' . $nextrun);
```

Comment:
```suggestion
			$symfonyStyle->note('Job: ' . $job->element . ' has run ' . $runned . ' times' . $nextrun);
```
"Run" is irregular verb, past particple is "run" and not "runned". Here to use past perfect because action has started in past and finished in past, therefore "has run".

Reply:
@brianteeman Please confirm.